napari/plugins/utils.py

```diff
@@ -82,8 +82,12 @@ def _get_preferred_readers(path: str) -> Iterable[Tuple[str, str]]:
     filtered_preferences : Iterable[Tuple[str, str]]
         Filtered patterns and their corresponding readers.
     """
-    reader_settings = get_settings().plugins.extension2reader
 
+    if osp.isdir(path):
+        path = osp.basename(path)
+        if not path.endswith('/'):
+            path = path + '/'
```

Comment:
I'm pretty sure that if this is tested, then it will fail on windows as the path separator is different on this system.